web-marketplace/src/lib/auth/auth.tsx

```diff
@@ -56,6 +59,35 @@ export const AuthProvider: React.FC<React.PropsWithChildren<unknown>> = ({
     account => account?.id === activeAccountId,
   );
 
+  const [profileBannerCard, setProfileBannerCard] = useAtom(
+    profileBannerCardAtom,
+  );
+
+  // hide the banner if a user has already set name, profile image, background image and one of the external links
+  useEffect(() => {
+    if (
+      !profileBannerCard[activeAccount?.id] &&
+      activeAccount?.name &&
+      activeAccount?.image &&
+      activeAccount?.bgImage &&
+      (activeAccount?.twitterLink || activeAccount?.websiteLink)
+    ) {
+      setProfileBannerCard({
+        ...profileBannerCard,
+        [activeAccount?.id]: true,
+      });
+    }
+  }, [
+    activeAccount?.bgImage,
+    activeAccount?.id,
+    activeAccount?.image,
+    activeAccount?.name,
+    activeAccount?.twitterLink,
+    activeAccount?.websiteLink,
+    profileBannerCard,
+    setProfileBannerCard,
+  ]);
```

Comment:
Seems like a long dependency array. Not a blocker, but any way to simplify this?

Reply:
I guess I could just extract the condition value in the `if` to some variable and have the useEffect depend on this variable to make it more readable